http-server-netty/src/main/java/io/micronaut/http/server/netty/HttpContentProcessor.java

```diff
@@ -15,18 +15,57 @@
  */
 package io.micronaut.http.server.netty;
 
-import io.micronaut.core.async.publisher.Publishers;
+import io.micronaut.core.type.Argument;
 import io.micronaut.core.util.Toggleable;
 import io.netty.buffer.ByteBufHolder;
-import org.reactivestreams.Subscriber;
+
+import java.util.Collection;
 
 /**
- * A reactive streams {@link org.reactivestreams.Processor} that processes incoming {@link ByteBufHolder} and
- * outputs a given type.
+ * This class represents the first step of the HTTP body parsing pipeline. It transforms
+ * {@link ByteBufHolder} instances that come from a
+ * {@link io.micronaut.http.netty.stream.StreamedHttpRequest} into parsed objects, e.g. json nodes
+ * or form data fragments.<br>
+ * Processors are stateful. They can receive repeated calls to {@link #add} with more data,
+ * followed by a call to {@link #complete} to finish up. Both of these methods accept a
+ * {@link Collection} {@code out} parameter that is populated with the processed items.
  *
- * @param <T> The type
  * @author Graeme Rocher
  * @since 1.0
  */
-public interface HttpContentProcessor<T> extends Publishers.MicronautPublisher<T>, Subscriber<ByteBufHolder>, Toggleable {
+public interface HttpContentProcessor extends Toggleable {
```

Comment:
add `@Internal`

Reply:
it's used in the jackson xml module, it should not be internal